cmd/ctr-remote/commands/optimize.go

```diff
@@ -247,33 +248,48 @@ func convert(clicontext *cli.Context, srcRef, dstRef name.Reference, runopts ...
 }
 
 // The order of the "in" list must be base layer first, top layer last.
-func optimize(clicontext *cli.Context, in []regpkg.Layer, config imgpkg.Image, opts ...sampler.Option) (out []regpkg.Layer, err error) {
-	root := ""
+func optimize(clicontext *cli.Context, in []regpkg.Layer, config imgpkg.Image, opts ...sampler.Option) (out []regpkg.Layer, done func(), err error) {
+	// Setup temporary workspace
+	tmpRoot, err := ioutil.TempDir("", "optimize")
+	if err != nil {
+		return nil, nil, err
+	}
+	if err = os.Chmod(tmpRoot, 0777); err != nil {
+		return nil, nil, err
+	}
+	defer os.RemoveAll(tmpRoot)
 	mktemp := func() (path string, err error) {
-		if path, err = ioutil.TempDir(root, "optimize"); err != nil {
+		if path, err = ioutil.TempDir(tmpRoot, "optimize"); err != nil {
 			return "", err
 		}
 		if err = os.Chmod(path, 0777); err != nil {
 			return "", err
 		}
 		return path, nil
 	}
-	if root, err = mktemp(); err != nil {
-		return nil, err
-	}
-	defer os.RemoveAll(root)
 
 	// mount layer loggers on temp directories.
 	var (
-		eg        errgroup.Group
-		lowerdirs []string
-		addLayer  = make([](func([]regpkg.Layer) ([]regpkg.Layer, error)), len(in))
+		eg         errgroup.Group
+		lowerdirs  []string
+		addLayer   = make([](func([]regpkg.Layer) ([]regpkg.Layer, error)), len(in))
+		layerFiles []*os.File
 	)
+	done = func() {
+		for _, f := range layerFiles {
+			if err := f.Close(); err != nil {
+				fmt.Printf("Warn: failed to close tmpfile %v: %v\n", f.Name(), err)
```

Comment:
consider use logrus

Reply:
I'll open a following-up PR for refactoring around logging of ctr-remote command.